### DISCUSSION TITLE
Suggested Features - Weighted List Items & Other Data Patterns

### DISCUSSION THREAD

**Discussion by Claggett** [benkeen/generatedata#768] [Ideas]
I really appreciate this tool! I teach a database course and I use this to generate fresh data sets I can import as tables into MySQL for my students to query.

I may be an odd case, but I'd love a way to weight occurrences of values. Right now I do a lot of creative things with the list data types as a workaround. For example, right now I'm generating a list of fictional healthcare providers, and I'd like to include Physicians, Nurses, and Therapists, but I want the therapists to be a rare case. What I do now is customize a "choose exactly 1 from list" and in the options, I put multiple instances of physician, multiples of nurse, and only one of therapist. It'd be lovely if the interface would give this type of control to state weights of values in a list.

Similarly, instead of just a random number range, being able to provide a range and an average and a standard deviation target for the generated sample to (come close to) fitting would be amazing. 

And my last wish list item would be to generate data with provided correlation weights between columns. It would be amazing to be able to hide data patterns for students to find through their queries and analysis.

Again - thanks for making the tool! I'm happy using it as is, but just wanted to write down my wish list in case any of these were possible for future iterations.

**Comment by benkeen** [comment]
Thanks @Claggett! Sorry for not responding sooner. 

> I may be an odd case, but I'd love a way to weight occurrences of values

Terrific idea, I could well imagine that being useful. That actually sounds like a fun (and short) project - which is a nice change from my recent fare. I'll see what I can rustle up!

> Similarly, instead of just a random number range, being able to provide a range and an average and a standard deviation target for the generated sample to (come close to) fitting would be amazing.

Sounds interesting, but pretty sure I'd need help with the math there... I suffered through creating the Normal Distribution data type and fear returning the code. :) 

> And my last wish list item would be to generate data with provided correlation weights between columns. It would be amazing to be able to hide data patterns for students to find through their queries and analysis.

Could you explain this one a bit more?

**Comment by benkeen** [comment]
The _Weighted List_ Data Type has now been added, btw.
